nilearn/_utils/niimg_conversions.py

```diff
@@ -193,6 +207,30 @@ def check_niimg(niimg, ensure_ndim=None, atleast_4d=False, dtype=None,
     """
     from ..image import new_img_like  # avoid circular imports
 
+    if isinstance(niimg, _basestring):
+        if (wildcards and
+            ni.EXPAND_PATH_WILDCARDS):
+            filenames = sorted(glob.glob(niimg))  # Ascending sorting
+
+            # processing filenames matching globbing expression
+            if len(filenames) > 1:
+                niimg = filenames  # iterable case
+            elif len(filenames) == 1:
+                # Only one file matching => loading it as is
+                niimg = filenames[0]
+            elif glob.has_magic(niimg):
+                # No files matching the glob expression, warn the user
+                message = ("No files matching the entered niimg expression: "
+                            "'%s'.\n You may have left wildcards usage "
+                           "activated: please set the global constant "
+                           "'EXPAND_PATH_WILDCARDS' to False to deactivate "
```

Comment:
As this message could be raised in the user console and wildcards=False is used internally (check_niimg should not be called directly by users), I would stick to the global constant in the message.

This makes me think that the wildcards option is useless... and should be removed from the function signature.